platform-api/src/services/cf_client.py

```
@@ -0,0 +1,125 @@
+            raise HTTPException(502, f"Cloudflare GraphQL API error: {resp.status_code}")
+
+        try:
+            # Parse the deeply nested GraphQL response
+            data = resp.json()["data"]["viewer"]["accounts"][0]["analyticsEngineEventsAdaptiveGroups"]
+            # Map it to a clean timeseries list
+            return [{"datetime": d["dimensions"]["datetimeHour"], "value": d["sum"]["double1"]} for d in data]
+        except (KeyError, IndexError) as e:
+            # Log the full response and exception for debugging
+            import logging
+            logger = logging.getLogger(__name__)
+            logger.error(f"Failed to parse Cloudflare Analytics response: {e}, status={resp.status_code}, body={resp.text}")
```

Ruff (0.15.11)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)